loopchain/blockchain/transactions/transaction_verifier.py

```diff
@@ -1,15 +1,44 @@
 from abc import ABC, abstractmethod
 from typing import TYPE_CHECKING
-from loopchain.crypto.signature import SignVerifier
-from loopchain.crypto.hashing import build_hash_generator
+
 from loopchain.blockchain.exception import TransactionDuplicatedHashError, TransactionInvalidHashError
 from loopchain.blockchain.exception import TransactionInvalidSignatureError
-
+from loopchain.crypto.hashing import build_hash_generator
+from loopchain.crypto.signature import SignVerifier
 
 if TYPE_CHECKING:
     from loopchain.blockchain.transactions import Transaction, TransactionVersioner
 
 
+def cache_result(tv_func):
+    """Cache verified result to Transaction.
+
+    If Transaction verified successfully, cache its result.
+    If Transaction verified failed with exception, cache its exception.
+    If Transaction didn't verified at all, its attribute may not exist.
+    """
+
+    def _wrapper(*args, **kwargs):
+        attr_name = "_cache_" + tv_func.__name__
+        tx: Transaction = args[1]
+        cached_result = getattr(tx, attr_name, False)
+
+        if isinstance(cached_result, Exception):
+            raise cached_result
```

Comment:
This doesn't consider storing exceptions.

Reply:
Tried to fix it. Please check.